### PR TITLE
Okta: Feature - 2 new actions

### PR DESCRIPTION
**Pull request by kelby-shelton**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

#### Added 2 new actions
- Added 'add_user_to_group'
- Added 'remove_user_from_group'
#### Minor code fixes
- Changed process_empty_response to recognize the same range of status codes as process_json_response and process_html_response

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by pdesai-crest** [comment]
Hey @kelby-shelton i've picked up this PR and will be actively working on it and merge this PR. Feel free to reach me out. 😁

File: readme.html [hunk 1]
```diff
@@ -181,5 +177,24 @@ <h2>Permissions</h2>
             </table>
         </p>
     </p>
-</body>
-</html>
+<p>Port Information</p>
+<p>
+    The app uses HTTP/ HTTPS protocol for communicating with the Zscaler server. Below are the default ports used by Splunk SOAR.
```
Comment (achandya-crest) [inline readme.html:182]: Add Okta server in place of Zscaler server.

File: okta_connector.py [hunk 2]
```diff
@@ -978,6 +1110,11 @@ def initialize(self):
             return self.set_status(phantom.APP_ERROR, "Error occurred while getting the Phantom server's Python major version.")
 
         self._state = self.load_state()
+        if not isinstance(self._state, dict):
+            self.debug_print("Resetting the state file with the default format")
+            self._state = {"app_version": self.get_app_json().get("app_version")}
+            return self.set_status(phantom.APP_ERROR, OKTA_STATE_FILE_CORRUPT_ERR)
```
Comment (achandya-crest) [inline okta_connector.py:1116]: return statement can be removed as no data is stored in the state file except app_version

**Review by achandya-crest — APPROVED** [review]

File: readme.html (hunk 1 above)
Comment (pdesai-crest) [reply readme.html:182]: Fixed

File: okta_connector.py (hunk 2 above)
Comment (pdesai-crest) [reply okta_connector.py]: Fixed

**Review by pdesai-crest — APPROVED** [review]

**Review by paulnawat — APPROVED** [review]
lgtm. left one minor comment

File: okta.json [hunk 3]
```diff
@@ -10,14 +10,14 @@
...
+    "min_phantom_version": "5.2.0",
     "product_name": "Okta",
     "app_wizard_version": "1.0.0",
-    "app_version": "2.2.5",
+    "app_version": "2.3.0",
     "python_version": "3",
     "fips_compliant": true,
     "latest_tested_versions": [
-        "Cloud, Version 2021.11.2"
+        "Cloud, Version 2022.05.3 C"
```
[the remark this answers is not in the document]
Comment (pdesai-crest) [reply okta.json:20]: no, it is actual version name

**Pull request merged**